pkg/scaffold/input/input.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Modified from github.com/kubernetes-sigs/controller-tools/pkg/scaffold/input/input.go
+
+package input
+
+// IfExistsAction determines what to do if the scaffold file already exists
+type IfExistsAction int
+
+const (
+	// Overwrite truncates and overwrites the existing file (default)
+	Overwrite IfExistsAction = iota
+
+	// Error returns an error and stops processing
+	Error
+
+	// Skip skips the file and moves to the next one
+	Skip
+)
+
+// Input is the input for scaffoldig a file
+type Input struct {
+	// Path is the file to write
+	Path string
```

Comment:
This is actually the path of the file, relative to the project root.
We should rename and highlight this to be more clear since its important that this is always the relative path of the file.
Update the comment to clarify this.
But I think we should rename this to be more obvious. Something like`PathRelativeToRoot` or `RelativePath`.

Reply:
My bad, this is actually converted to the absolute path down below in `doFile()`
```Go
// Ensure path is set correctly.
i.Path = filepath.Join(s.ProjectPath, i.Path)
```
https://github.com/operator-framework/operator-sdk/pull/581/files#diff-d421d187b143814e028a5d9df0266011R108

And then used to write the actual file:
```Go
f, err := s.GetWriter(i.Path, mode)
```

So it starts out as the relative path but turns into the absolute path. That's slightly confusing.
We should probably have two fields `i.RelativePath` and `i.AbsPath`.

But we can do that in a follow up to clean up and make the fields in scaffold.go and input.go more readable.

For now just update the comment that it should be set to the relative path.

Reply:
I would think that it should be a local variable and passed to the `doTemplate` function. We never use the value other than in that call stack, I would prefer to not add export fields on a struct for what is effectively a local variable IMO.

Reply:
Changed such that `i.Path` is not modified.